File: AdventOfCode2025/Day1_SecretEntrance/solution_part1.go
package main

// import (
// 	"bufio"
// 	"fmt"
// 	"os"
// 	"strconv"
// )

// const RoundTurnValue = 100

// var currentValue = 50
// var zeroCount = 0

// func main2() {
// 	file, err := os.Open("input.txt")
// 	if err != nil {
// 		fmt.Println("Error opening file:", err)
// 		return
// 	}
// 	defer file.Close()

// 	scanner := bufio.NewScanner(file)
// 	if err := scanner.Err(); err != nil {
// 		fmt.Println("Error reading file:", err)
// 	}

// 	for scanner.Scan() {
// 		line := scanner.Text()
// 		dir := line[0] // First character is the direction
// 		if dir != 'L' && dir != 'R' {
// 			fmt.Println("Invalid character in line:", dir)
// 			return
// 		}

// 		dist, err := strconv.Atoi(line[1:]) // The rest of the string is the number -> parse to int
// 		if err != nil {
// 			fmt.Println("Invalid number in line:", line)
// 			continue
// 		}

// 		diff := dist % RoundTurnValue
// 		if dir == 'L' {
// 			diff *= -1
// 		} else {
// 			diff *= 1
// 		}

// 		currentValue = (currentValue + diff) % RoundTurnValue
// 		if currentValue == 0 {
// 			zeroCount++
// 		}
// 	}
// 	fmt.Println(zeroCount)
// }

File: AdventOfCode2025/Day7_Laboratories/solution_part1.py
split_count = 0
f = open("input.txt", "r")
prev_row = {f.readline().find("S")}
line = f.readline()
while line:
    curr_row = set()
    for _, j in enumerate(prev_row):
        if (line[j] == "^"):
            curr_row.add(j - 1)
            curr_row.add(j + 1)
            split_count += 1
        else :
           curr_row.add(j) 
    prev_row = curr_row
    line = f.readline()    

f.close()
print(split_count)
